src/WarcraftLegacies.Source/Mechanics/Scourge/HelmOfDominationDropsWhenScourgeLeaves.cs

```diff
@@ -1,22 +1,25 @@
-// using WarcraftLegacies.Source.Legends;
-// using WarcraftLegacies.Source.Setup;
-// using WarcraftLegacies.Source.Setup.FactionSetup;
-//
-// using static War3Api.Common;  using static MacroTools.GeneralHelpers;
-// {
-//   public class HelmOfDominationDropsWhenScourgeLeaves{
-//
-//     private static void Actions( ){
-//       if (GetTriggerFaction() == ScourgeSetup.FactionScourge && ArtifactSetup.ArtifactHelmofdomination.OwningUnit == LegendScourge.LegendLichking.Unit){
-//         SetItemPosition(ArtifactSetup.ArtifactHelmofdomination.Item, GetRectCenterX(gg_rct_LichKing), GetRectCenterY(gg_rct_LichKing));
-//       }
-//     }
-//
-//     public static void Setup( ){
-//       trigger trig = CreateTrigger();
-//       OnFactionGameLeave.register(trig);
-//       TriggerAddAction(trig,  Actions);
-//     }
-//
-//   }
-// }
+using WarcraftLegacies.Source.Setup.Legends;
+using WarcraftLegacies.Source.Setup;
+using WarcraftLegacies.Source.Setup.FactionSetup;
+using static War3Api.Common;
+using MacroTools.FactionSystem;
+using MacroTools.Extensions;
+
+public class HelmOfDominationDropsWhenScourgeLeaves
+{
+    public static void Setup()
+    {
+        Faction.GameLeave += Faction_GameLeave;
+    }
+
+    private static void Faction_GameLeave(object? sender, Faction e)
```

Comment:
I don't know if it's standard for C# events, but lets' call the Faction parameter something descriptive like `leavingFaction`.

Reply:
Microsoft usually doesn't name the parameter anything but e. I think they expect people to look at the type directly. 
But I will follow your preferred style of course.